cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -4028,6 +4028,48 @@
     </description>
   </property>
 
+  <!-- Retry configuration for spark metrics retrieval -->
+  <property>
+    <name>spark.metrics.strategy.retry.policy.base.delay.ms</name>
+    <value>5000</value>
+    <description>
+      The base delay between retries in milliseconds
+    </description>
+  </property>
+
+  <property>
+    <name>spark.metrics.strategy.retry.policy.max.delay.ms</name>
+    <value>10000</value>
+    <description>
+      The maximum delay between retries in milliseconds
+    </description>
+  </property>
+
+  <property>
+    <name>spark.metrics.strategy.retry.policy.max.retries</name>
+    <value>3</value>
```

Comment:
You can set it to higher value, such as 10 since we should lower the base delay.